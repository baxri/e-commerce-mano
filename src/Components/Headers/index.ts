import MainHeader from "./Main";

export { MainHeader };
