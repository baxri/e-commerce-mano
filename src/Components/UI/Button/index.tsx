import React from "react";
// @ts-ignore
import { pick } from "@styled-system/props";

import { Column, Row } from "../View";
import { Container, ContainerProps, Indicator, Inner, Title } from "./styles";

type Props = ContainerProps & {
  disabled?: boolean;
  icon?: () => React.ReactNode;
  link?: boolean;
  loading?: boolean;
  secondary?: boolean;
  error?: boolean;
  title?: string;
  primary?: boolean;
  onPress?: () => void;
};

function Button({
  disabled = false,
  link = false,
  icon,
  title = "",
  loading = false,
  secondary = false,
  primary = false,
  error = false,
  onPress,
  ...rest
}: Props) {
  return (
    <Container
      {...pick(rest)}
      activeOpacity={0.8}
      disabled={disabled || loading}
      onPress={onPress}
    >
      {({ pressed }: { pressed: boolean }) => (
        <Inner
          disabled={disabled}
          error={error}
          link={link}
          pressed={pressed}
          primary={primary}
          secondary={secondary}
        >
          {loading ? (
            <Indicator error={error} link={link} secondary={secondary} />
          ) : (
            <Row>
              <Column mr={title && icon ? 5 : 0} alignCenter justifyCenter>
                {icon && icon()}
              </Column>
              <Title
                disabled={disabled}
                error={error}
                link={link}
                numberOfLines={2}
                pressed={pressed}
                secondary={secondary}
              >
                {title}
              </Title>
            </Row>
          )}
        </Inner>
      )}
    </Container>
  );
}

export default Button;
