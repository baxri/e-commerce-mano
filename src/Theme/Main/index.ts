import button from "./button";
import colors from "./colors";
import common from "./common";
import fonts from "./fonts";

export default {
  name: "Main Theme",
  dark: false,
  button,
  common,
  colors,
  fonts,

  // --   0  1  2  3  4   5   6   7   8   9
  space: [0, 2, 4, 8, 12, 16, 20, 24, 32, 64],
  // --       0   1   2   3    4   5   7   8   9
  fontSizes: [10, 12, 14, 16, 18, 20, 22, 24, 26],
  // --         0   1   2   3    4   5   7   8   9
  lineHeights: [13, 15, 17, 19, 21, 23, 25, 27, 29],
  // --         0    1    2    3
  fontWeights: [400, 500, 600, 700],
};
