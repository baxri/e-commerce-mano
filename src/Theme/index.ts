import MainTheme from "./Main";

enum ThemeID {
  Main = "main",
}

const THEMES: Record<ThemeID, any> = {
  [ThemeID.Main]: MainTheme,
};

export { ThemeID, THEMES };
