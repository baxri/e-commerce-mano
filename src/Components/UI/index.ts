import BottomTabBar from "./BottomTabBar";
import Button from "./Button";
import FlatList from "./FlatList";
import KeyboardSpacer from "./KeyboardSpacer";
import ScrollView from "./ScrollView";
import Spinner from "./Spinner";
import Text from "./Text";
import { Column, Row } from "./View";

export {
  Text,
  Button,
  Column,
  KeyboardSpacer,
  Row,
  ScrollView,
  Spinner,
  FlatList,
  BottomTabBar,
};
