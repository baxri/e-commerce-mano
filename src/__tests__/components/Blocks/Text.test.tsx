import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';

import { ThemeID, THEMES } from 'Theme';

import { Text } from 'Components/UI';

test('Text', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={THEMES[ThemeID.Main]}>
        <Text>TEST</Text>
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
