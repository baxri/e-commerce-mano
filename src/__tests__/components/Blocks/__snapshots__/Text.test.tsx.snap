// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text 1`] = `
<Text
  allowFontScaling={false}
  color="black"
  fontFamily="Inter"
  fontSize={2}
  fontWeight={0}
  style={
    [
      {
        "color": "#0F1120",
        "fontFamily": "Inter",
        "fontSize": 14,
        "fontWeight": "400",
      },
    ]
  }
>
  TEST
</Text>
`;
